### PR TITLE
Reduce guava footprint

### PR DESCRIPTION
**Pull request by car-roll**
Reduce the use of Guava from the plugin in order to be compatible with all future Jenkins versions where Guava may be bumped or not exposed.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesProvisioningLimits.java [hunk 1]
```diff
@@ -121,13 +120,11 @@ public void unregister(@Nonnull KubernetesCloud cloud, @Nonnull PodTemplate podT
     }
 
     @Nonnull
-    @VisibleForTesting
```
Comment (Vlatombe) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesProvisioningLimits.java:124]: Can we keep something like // VisibleForTesting ? This has still documentation value.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java [hunk 2]
```diff
@@ -75,7 +72,17 @@
...
-    public static final HashFunction LABEL_DIGEST_FUNCTION = Hashing.sha1();
+    protected static final MessageDigest LABEL_DIGEST_FUNCTION;
+    static {
+        try {
+            LABEL_DIGEST_FUNCTION = MessageDigest.getInstance("SHA-256");
+        } catch (NoSuchAlgorithmException e) {
+            // will never happen, SHA-256 support required on every Java implementation
+            e.printStackTrace();
+            // throw runtime exception to allow variable to be set as final
+            throw new RuntimeException(e);
```
Comment (Vlatombe) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java:83]: Why not `throw new AssertionError(e);` like the previous impl?

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/StandardPlannedNodeBuilder.java [hunk 3]
```diff
@@ -25,10 +26,11 @@
...
                     .build();
             displayName = agent.getDisplayName();
-            f = Futures.immediateFuture(agent);
-        } catch (IOException | Descriptor.FormException e) {
+            f = CompletableFuture.completedFuture(agent);
+        } catch (IOException | Descriptor.FormException | NoSuchAlgorithmException e) {
             displayName = null;
-            f = Futures.immediateFailedFuture(e);
+            f = new CompletableFuture();
+            ((CompletableFuture<?>) f).completeExceptionally(e);
```
Comment (Vlatombe) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/StandardPlannedNodeBuilder.java:33]: f can be initialized as a CompletableFuture so you avoid this cast.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesProvisioningLimits.java (hunk 1 above)
Comment (car-roll) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesProvisioningLimits.java:124]: Just wondering if there was an alternative java annotation or access modifier we can use?

Comment (Vlatombe) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesProvisioningLimits.java:124]: Well, we could maybe use something like `@Restricted(NoExternalUse.class)`

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/StandardPlannedNodeBuilder.java (hunk 3 above)
Comment (car-roll) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/StandardPlannedNodeBuilder.java]: fixed in ff1d11f

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java (hunk 2 above)
Comment (car-roll) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java]: fixed in ff1d11f

**Comment by Vlatombe** [comment]
@car-roll they should be passing in general. There can be some flakes due to infrastructure.
I checked the jackson failures (ContainerExecDecoratorTest) in this pr and they're not in master, so this seems like a regression.

File: src/test/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesTestUtil.java [hunk 4]
```diff
@@ -181,7 +176,9 @@ public static boolean isWindows() {
      * Labels to add to the pods so we can match them to a specific build run, test class and method
      */
     public static Map<String, String> getLabels(KubernetesCloud cloud, Object o, TestName name) {
-        HashMap<String, String> l = Maps.newHashMap(DEFAULT_LABELS);
+        Map<String, String> l = new HashMap<>();
+        l.put(BRANCH_NAME, BRANCH_NAME == null ? "undefined" : BRANCH_NAME);
+        l.put(BUILD_NUMBER, BUILD_NUMBER == null ? "undefined" : BUILD_NUMBER);
```
Comment (Vlatombe) [inline src/test/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesTestUtil.java:181]: This should fix the jackson problem.
```suggestion
        l.put("BRANCH_NAME", BRANCH_NAME == null ? "undefined" : BRANCH_NAME);
        l.put("BUILD_NUMBER", BUILD_NUMBER == null ? "undefined" : BUILD_NUMBER);
```

**Review by Vlatombe — APPROVED** [review]

**Comment by car-roll** [comment]
@Vlatombe thanks!  Just a heads up, I don't think release drafter picked up this PR in the release notes?

**Comment by Vlatombe** [comment]
Don't worry, it did.

**Comment by MRamonLeon** [comment]
@Vlatombe could you release it please? Or do you know when it may be released? thanks!

**Pull request merged**